### DISCUSSION TITLE
QTree throws TypeError: Cannot read property 'map' of undefined

### DISCUSSION THREAD

**Discussion by vijayganeshpk** [quasarframework/quasar#10064] [General - Components / Directives / etc] [answered]
**Describe the bug**
For a given tree data structure QTree is throwing an Error ("`TypeError: Cannot read property 'map' of undefined`")

**Codepen/jsFiddle/Codesandbox (required)**
https://codesandbox.io/s/vibrant-clarke-pmipj?file=/src/pages/Index.vue

**To Reproduce**
1. Go to line 551 in the above codesandbox.io sample
2. Comment line 551 `const treeData = ref(responseWorking.payload)` and uncomment line 552 `const treeData = ref(responseNotWorking.payload)`
3. Save, refresh the output and goto "Console"

**Expected behavior**
The new child node is shown in the tree

**Screenshots**

**Platform (please complete the following information):**
Quasar Version: 2.0.2
@quasar/app Version: 3.0.2

Quasar mode:
  - [ X ] SPA
  - [ ] SSR
  - [ ] PWA
  - [ ] Electron
  - [ ] Cordova
  - [ ] Capacitor
  - [ ] BEX

Tested on:
  - [ X ] SPA
  - [ ] SSR
  - [ ] PWA
  - [ ] Electron
  - [ ] Cordova
  - [ ] Capacitor
  - [ ] BEX

OS: macOS Big Sur v11.4
Node: v14.17.3
NPM: 6.14.13
Yarn: 1.22.10
Browsers: Vivaldi (4.0.2312.38 (Stable channel) (x86_64))
iOS:
Android:
Electron:

**Additional context**
The only difference is the content for children node in the last leaf:

working node

```json
{
  "children": [],
  "id": 9,
  "label": 9
}
```

Not working node

```json
{
  "children": [
    {
      "id": 300,
      "label": "9.300",
      "location_id": 7,
      "type_id": 8
    }
  ],
  "id": 9,
  "label": 9
}
```

**Comment by pdanpdan** [comment, marked as answer]
What is obvious from a quick view is that you have duplicate id 2 in the not working

**Reply by vijayganeshpk** [reply]
Thank you, I was combining two tables for the parent and child (master detail) and had missed the most obvious thing to check.
